internal/resources/sample_bucket.go

```diff
@@ -0,0 +1,351 @@
+func NewSampleBucket() resource.Resource {
+	return &SampleBucket{}
+}
+
+// Metadata returns the samples resource type name.
+func (s *SampleBucket) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_samplebucket"
+}
+
+// Configure It adds the provider configured api to the project resource.
+func (s *SampleBucket) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	s.Data = data
+}
+
+// ImportState imports a remote sample cluster that is not created by Terraform.
+func (s *SampleBucket) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+// Schema defines the schema for the samples resource.
+func (s *SampleBucket) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = SampleBucketSchema()
+}
+
+func (s *SampleBucket) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.Bucket
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	BucketRequest := samplebucket.CreateSampleBucketRequest{
+		Name: plan.Name.ValueString(),
+	}
+	if err := s.validateCreateBucket(plan); err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating bucket",
+			"Could not create bucket, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	var organizationId = plan.OrganizationId.ValueString()
+	var projectId = plan.ProjectId.ValueString()
+	var clusterId = plan.ClusterId.ValueString()
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/sampleBuckets", s.HostURL, organizationId, projectId, clusterId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusCreated}
+	response, err := s.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		BucketRequest,
+		s.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating bucket",
+			errorMessageWhileBucketCreation+api.ParseError(err),
+		)
+		return
+	}
+
+	BucketResponse := samplebucket.CreateSampleBucketResponse{}
+	err = json.Unmarshal(response.Body, &BucketResponse)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating bucket",
+			errorMessageWhileBucketCreation+"error during unmarshalling: "+err.Error(),
+		)
+		return
+	}
+
+	plan.Id = types.StringValue(BucketResponse.Id)
+	diags = resp.State.Set(ctx, plan)
```

Comment:
We need to set all computed fields in the plan to null. Additionally, if a field is both optional and computed, and no value is provided, we should populate it with null. We can refer to other resources for guidance.
i.e. initializeAllowListWithPlanAndId